File: frontend/src/services/admin/profileService.ts
import api from "@/lib/axios";
import type {
  ChangePasswordPayload,
  UpdateProfilePayload,
} from "@/types/admin/profile";

export const profileService = {
  getMyProfile: async () => {
    const res = await api.get("/profile");
    return res.data;
  },

  updateMyProfile: async (payload: UpdateProfilePayload) => {
    const res = await api.put("/profile", payload);
    return res.data;
  },

  changePassword: async (payload: ChangePasswordPayload) => {
    const res = await api.put("/profile/change-password", payload);
    return res.data;
  },
};

File: backend/src/main/java/com/javaweb/service/admin/ProfileService.java
package com.javaweb.service.admin;

import com.javaweb.model.request.profile.ChangePasswordRequestDTO;
import com.javaweb.model.request.profile.UpdateProfileRequestDTO;
import com.javaweb.model.response.profile.ProfileResponseDTO;

public interface ProfileService {
  ProfileResponseDTO getMyProfile();
  void updateMyProfile(UpdateProfileRequestDTO request);
  void changePassword(ChangePasswordRequestDTO request);
}